internal/lockclient/simple_client.go

```diff
@@ -103,3 +133,91 @@ func (sc *SimpleClient) Release(d lockservice.Descriptors) error {
 func (sc *SimpleClient) StartService(cfg Config) error {
 	panic("TODO")
 }
+
+// Checkacquire checks for acquisition of lock and returns the owner if the lock
+// is already acquired.
+// The errors returned can be due to HTTP errors or marshalling errors.
+// A "file is not acquired" error is returned if so and no error and an owner is
+// returned if the object is acquired.
+func (sc *SimpleClient) Checkacquire(d lockservice.ObjectDescriptor) (string, error) {
+	if sc.cache != nil {
+		owner, err := sc.getFromCache(d)
+		if err != nil {
+			return "", err
+		}
+		return owner, nil
+	}
+
+	endPoint := sc.config.IPAddr + ":" + sc.config.PortAddr + "/checkacquire"
+	data := lockservice.LockCheckRequest{FileID: d.ObjectID}
+	requestJSON, err := json.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+
+	req, err := http.NewRequest("POST", endPoint, bytes.NewBuffer(requestJSON))
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	if resp.StatusCode != 200 {
+		return "", lockservice.Error(strings.TrimSpace(string(body)))
+	}
+
+	var ownerData lockservice.CheckacquireRes
+	err = json.Unmarshal(body, &ownerData)
+	if err != nil {
+		return "", err
+	}
+
+	return ownerData.Owner, nil
+}
+
+// getFromCache checks the lock status on the descriptor in the cache.
+// This function returns an error if the cache doesn't exist or the
+// file is NOT acquired.
+func (sc *SimpleClient) getFromCache(d lockservice.ObjectDescriptor) (string, error) {
+	if sc.cache != nil {
+		owner, err := sc.cache.GetElement(cache.NewSimpleKey(d.ObjectID, ""))
+		if err != nil {
+			return "", lockservice.ErrCheckacquireFailure
```

Comment:
ErrCheckacquireFailure is "file is not acquired". That doesn't seem relevant here.

Instead shouldn't the error being returned ineidate that the element doesn't exist in the cache?

Reply:
`CheckAcquire` is checking whether the `file` is acquired or not. Seems okay because we need to hide the cache and behave as it was a network call.

Reply:
Ah okay got it. The element will be in the cache only if it is acquired. Still, maybe a different error would be useful for debugging in the future?

Reply:
I dont know whether we should. Still think it should be transparent, not sure what future needs might be.